lib/crypto/src/elliptic-curve/keys.rs

```diff
@@ -0,0 +1,43 @@
+//! Key-handling related logic.
+
+use super::{
+    affine::AffinePoint, curve::PrimeCurve, projective::ProjectivePoint,
+};
+
+/// Elliptic curve public keys.
+///
+/// This is a wrapper type for [`AffinePoint`] which ensures an inner
+/// non-identity point and provides a common place to handle encoding/decoding.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PublicKey<C>
+where
+    C: PrimeCurve,
+{
+    point: AffinePoint<C>,
+}
+
+impl<C> PublicKey<C>
+where
+    C: PrimeCurve,
+{
+    /// Convert an [`AffinePoint`] into a [`PublicKey`].
+    pub fn from_affine(point: AffinePoint<C>) -> Self {
+        Self { point }
+    }
+
+    /// Compute a [`PublicKey`] from a secret scalar value (i.e. a secret key
+    /// represented as a raw scalar value). Returns `None` if the value computed
+    /// is the identity point.
+    pub fn from_secret_scalar(scalar: &C::Uint) -> Option<Self> {
+        (&ProjectivePoint::GENERATOR * scalar)
+            .to_affine()
+            .map(|point| Self { point })
```

Comment:
How about implementing `From` trait here? It is just a wrapper.

Reply:
Sadly, this doesn't have the same signature as either `From` (returns `Self`) or `TryFrom` (returns `Result<Self, ..>`.

The reason is that you can end up with the identity when converting from projective coordinates to affine coordinates, but we don't want to leak that to the user, so I decided to return `Option<Self>`.

This is not a strong opinion, I will probably end up returning a `Result<Self, ..>`, but I haven't figured out the best way to do this. If I go this route, I'll turn it into a `TryFrom` implementation.

Reply:
Sounds good.